### PR TITLE
Modified warn logs to debug logs when there is a mismatch in callback URL

### PR DESCRIPTION
**Pull request by randiepathirage**
# Purpose

In an authorization code grant login when there is a mismatch in callback URL, the error message does not give a clear understanding about what's the warning.

Since this not a server error it should not be printed in error mode.

To fix this, Modified warn logs to debug logs when there is a mismatch in callback URL and changed it to a more meaningful message.

Related Issue : [wso2/product-is#6201](https://github.com/wso2/product-is/issues/6201)

### CODE REVIEW

File: .jpb/persistence-units.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  <component name="PersistenceUnitSettings">
+    <persistence-units>
+      <persistence-unit name="Default">
+        <packages>
+          <package value="" />
+        </packages>
+      </persistence-unit>
+    </persistence-units>
+  </component>
+</project>
```
Comment (mefarazath) [inline .jpb/persistence-units.xml:12]:
```suggestion
</project>

```

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java [hunk 2]
```diff
@@ -197,7 +197,9 @@ public OAuth2ClientValidationResponseDTO validateClientInfo(String clientId, Str
                 validationResponseDTO.setPkceSupportPlain(appDO.isPkceSupportPlain());
                 return validationResponseDTO;
             } else {    // Provided callback URL does not match the registered callback url.
-                log.warn("Provided Callback URL does not match with the provided one.");
+                if (log.isDebugEnabled()) {
+                    log.debug("Provided Callback URL: " + callbackURI + " does not match with the provided one.");
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java:201]: shouldn't the message say, provided callback does not match with registered ones?

File: .jpb/persistence-units.xml (hunk 1 above)
Comment (piraveena) [reply .jpb/persistence-units.xml:12]: Can you explain why this xml file is added?

**Comment by randiepathirage** [comment]
Closing this PR. Created a new PR with this changes. 
https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1753

**Pull request closed without merging**